test/openssl/test_fips.rb

```diff
@@ -4,12 +4,37 @@
 if defined?(OpenSSL)
 
 class OpenSSL::TestFIPS < OpenSSL::TestCase
+  def test_fips_mode_get_is_true_on_fips_mode_enabled
+    unless ENV["CI"] && ENV["TEST_RUBY_OPENSSL_FIPS_ENABLED"]
+      omit "Only for on FIPS mode environment on CI"
+    end
+
+    assert_separately([{ "OSSL_MDEBUG" => nil }, "-ropenssl"], <<~"end;")
+      assert OpenSSL.fips_mode == true, ".fips_mode returns true on FIPS mode enabled"
+    end;
+  end
+
+  def test_fips_mode_get_is_false_on_fips_mode_disabled
+    unless ENV["CI"] && !ENV["TEST_RUBY_OPENSSL_FIPS_ENABLED"]
+      omit "Only for non-FIPS mode environment on CI"
+    end
+
+    assert_separately([{ "OSSL_MDEBUG" => nil }, "-ropenssl"], <<~"end;")
+      assert OpenSSL.fips_mode == false, ".fips_mode returns false on FIPS mode disabled"
+    end;
+  end
+
   def test_fips_mode_is_reentrant
     OpenSSL.fips_mode = false
     OpenSSL.fips_mode = false
   end
 
-  def test_fips_mode_get
+  def test_fips_mode_get_with_fips_mode_set
+    if OpenSSL.fips_mode && openssl?(3, 0, 0)
+      pend('OpenSSL::OPENSSL_FIPS and fips_mode_set are not properly ' \
+           'implemented in OpenSSL 3')
+    end
+
```

Comment:
Do you intend to implement the `OpenSSL.fips_mode=` via `EVP_default_properties_enable_fips` or is it more complex change then the `OpenSSL.fips_mode` you propose above?

Reply:
I was able to implement the `OpenSSL.fips_mode=` and `OpenSSL::OPENSSL_FIPS` after some investigations on my local. I was able to implement the `OpenSSL.fips_mode=` without `OSSL_LIB_CTX *libctx` variable. The logic is simple. Now I am testing. I will rebase this PR!

Reply:
Rebased!